reactor-netty-core/src/main/java/reactor/netty/transport/TransportConnector.java

```diff
@@ -219,7 +220,31 @@ static void doConnect(SocketAddress remoteAddress, @Nullable Supplier<? extends
 				return monoChannelPromise;
 			}
 
+			if (config instanceof ClientTransportConfig) {
+				final ClientTransportConfig clientTransportConfig = (ClientTransportConfig)config;
+				if (clientTransportConfig.doOnResolve != null) {
+					clientTransportConfig.doOnResolve.accept(Connection.from(channel));
+				}
+			}
+
 			Future<SocketAddress> resolveFuture = resolver.resolve(remoteAddress);
+
+			if (config instanceof ClientTransportConfig) {
```

Comment:
What about checking that we have `doOnResolveError`/`doAfterResolve` and then adding a listener.
May be in the common cases (where we do not need to propagate something) these callbacks will not exist.